network/src/traits.rs

```diff
@@ -1,109 +1,13 @@
-    /// Sends a serialized message to a network destination
-    /// Implementations of this method must not block on I/O.
-    async fn unreliable_send_message(
-        &mut self,
-        address: Multiaddr,
-        message: BincodeEncodedPayload,
-    ) -> ();
-
-    async fn unreliable_send(&mut self, address: Multiaddr, message: &Self::Message) -> () {
-        let message =
-            BincodeEncodedPayload::try_from(message).expect("Failed to serialize payload");
-        self.unreliable_send_message(address, message).await
-    }
-
-    /// Broadcasts a message to all `addresses` passed as an argument.
-    /// The attempts to send individual messages are best effort and will not be retried.
-    async fn unreliable_broadcast(
-        &mut self,
-        addresses: Vec<Multiaddr>,
-        message: &Self::Message,
-    ) -> () {
-        let message =
-            BincodeEncodedPayload::try_from(message).expect("Failed to serialize payload");
-        for address in addresses {
-            // this is ok assuming implementations make unreliable_send_message non-blocking
-            self.unreliable_send_message(address, message.clone()).await
-        }
-    }
-}
-
-#[async_trait]
-pub trait LuckyNetwork: UnreliableNetwork {
-    fn rng(&mut self) -> &mut SmallRng;
-
-    /// Pick a few addresses at random (specified by `nodes`) and try (best-effort) to send the
-    /// message only to them. This is useful to pick nodes with whom to sync.
-    async fn lucky_broadcast(
-        &mut self,
-        mut addresses: Vec<Multiaddr>,
-        message: &Self::Message,
-        nodes: usize,
-    ) -> () {
-        addresses.shuffle(self.rng());
-        addresses.truncate(nodes);
-        self.unreliable_broadcast(addresses, message).await
-    }
-}
-
-#[async_trait]
-pub trait ReliableNetwork: BaseNetwork {
-    async fn send(
-        &mut self,
-        address: Multiaddr,
-        message: &Self::Message,
-    ) -> CancelOnDropHandler<MessageResult> {
-        let message =
-            BincodeEncodedPayload::try_from(message).expect("Failed to serialize payload");
-        self.send_message(address, message).await
-    }
-
-    async fn send_message(
-        &mut self,
-        address: Multiaddr,
-        message: BincodeEncodedPayload,
-    ) -> CancelOnDropHandler<MessageResult>;
-
-    async fn broadcast(
-        &mut self,
-        addresses: Vec<Multiaddr>,
-        message: &Self::Message,
-    ) -> Vec<CancelOnDropHandler<MessageResult>> {
-        let message =
-            BincodeEncodedPayload::try_from(message).expect("Failed to serialize payload");
-        let mut handlers = Vec::new();
-        for address in addresses {
-            let handle = self.send_message(address, message.clone()).await;
-            handlers.push(handle);
-        }
-        handlers
-    }
-}
 
 #[async_trait]
-pub trait UnreliableNetwork2<Message: Clone + Send + Sync> {
+pub trait UnreliableNetwork<Message: Clone + Send + Sync> {
```

Comment:
Would it be worth removing the `2` suffix now that these are the only one again? Or do we not really care?

Reply:
Yeah the old traits were for grpc networks and now that we've finished the conversion we can just rename it.